notificationworkerlambda/src/main/scala/com/gu/notifications/worker/AndroidSender.scala

```diff
@@ -35,16 +35,18 @@ class AndroidSender(val config: FcmWorkerConfiguration, val firebaseAppName: Opt
   logger.info(s"Using thread pool size: ${config.threadPoolSize}")
   logger.info(s"Topics for individual send: ${config.allowedTopicsForIndividualSend.mkString(",")}")
   logger.info(s"Concurrency for individual send: ${config.concurrencyForIndividualSend}")
+  logger.info(s"Concurrency for message processing: ${config.concurrencyForMessages}")
+  logger.info(s"HttpClient object pool: ${config.httpClientPoolSize}")
 
   override implicit val ioContextShift: ContextShift[IO] = IO.contextShift(ec)
   override implicit val timer: Timer[IO] = IO.timer(ec)
 
   val fcmFirebase: Try[FcmFirebase] = FcmFirebase(config.fcmConfig, firebaseAppName)
 
-  val fcmClients: Seq[Try[FcmClient]] = Seq.fill(20)(fcmFirebase).map(firebase => firebase.map(FcmClient(_)))
+  val fcmClients: Seq[Try[FcmClient]] = Seq.fill(config.httpClientPoolSize)(fcmFirebase).map(firebase => firebase.map(FcmClient(_)))
 
   val deliveryServiceStream: Stream[IO, Fcm[IO]] = 
-    Stream.emits(fcmClients).covary[IO].flatMap(_.fold(e => Stream.raiseError[IO](e), c => Stream.eval[IO, Fcm[IO]]( IO.delay(new Fcm(c)))))
+    Stream.emits(fcmClients).covary[IO].flatMap(_.fold(e => Stream.raiseError[IO](e), c => Stream.eval[IO, Fcm[IO]]( IO.pure(new Fcm(c)))))
```

Comment:
If we create the `IO` with `delay` method, the `new Fcm` will be executed every time we evaluate this `IO` object.  

Now that we want to evaluate this IO object to get the `Fcm` for every device token, it may be good not to create this `Fcm` instance for every device?  So I use `pure` here to create the `Fcm` instance and put this instance into this `IO` object.

The `Fcm` does not have any states actually, so it should be safe to reuse the same instance.

Reply:
Thanks for the explanation